cdap-common/src/main/java/io/cdap/cdap/common/internal/remote/RemoteClient.java

```diff
@@ -116,9 +118,16 @@ public HttpResponse execute(HttpRequest request) throws IOException, Unauthorize
     try {
       HttpResponse response = HttpRequests.execute(httpRequest, httpRequestConfig);
       int responseCode = response.getResponseCode();
-      if (Constants.RETRYABLE_HTTP_CODES.contains(responseCode)) {
+      // 503 is always retryable. Other 5xx errors are retryable if the request is idempotent (handled in
+      // RemoteClient#executeIdempotent(HttpRequest)
+      if (responseCode == HttpURLConnection.HTTP_UNAVAILABLE) {
         throw new ServiceUnavailableException(discoverableServiceName, response.getResponseBodyAsString());
       }
+      if (Constants.HTTP_SERVER_ERROR_CODES.contains(responseCode)) {
+        String message = String.format("Service %s is not unavailable: %s",
```

Comment:
Please correct the message.

Reply:
fixed "is not unavailable" -> "is unavailable"